### PR TITLE
fix filters when use the same key multiple times. closes #231

### LINKED ISSUE #231: Cannot apply two where directives
**Describe the bug**
I cannot apply two where directives when i use the same key.

**Expected behavior**
Apply two where clauses.

**Schema**

```graphql
type Campaign @model {
  id: ID! @globalId
  posts(
    since: Date @where(clause: "whereDate", key: "publish_at", operator: ">=")
    until: Date @where(clause: "whereDate", key: "publish_at", operator: "<=")
  ): [Post!]!
    @hasMany(type: "relay")
}
```

**Result query**

```sql
SELECT
    *
FROM (
    SELECT
        *
    FROM
        `shr_posts`
    WHERE
        date(`publish_at`) <= '2018-07-31'
    LIMIT 10 offset 0
) AS `posts`
```

**Environment**

Lighthouse Version: v2.2-beta.1
Laravel Version: 5.5
PHP Version: 7.2.7

### PR DESCRIPTION
**Pull request by hosmelq**
**Related Issue(s)**

#231 

**PR Type**

Bugfix

**Changes**

I create a composed key for the where directive based in the argument name and key name.

**Breaking changes**

None

### CODE REVIEW

File: src/Support/Database/QueryFilter.php [hunk 1]
```diff
@@ -61,6 +61,10 @@ public static function build($query, array $args)
     public function filter($query, array $args = [])
     {
         foreach ($this->filters as $key => $filter) {
+            if (strpos($key, '.')) {
```
Comment (spawnia) [inline src/Support/Database/QueryFilter.php]: Would add a comment here:

File: src/Support/Traits/HandlesQueryFilter.php [hunk 2]
```diff
@@ -17,11 +17,16 @@
      */
     protected function injectFilter(ArgumentValue $argument, array $filter)
     {
+        $argName = $argument->getArgName();
         $key = $this->queryFilterKey($argument);
         $query = QueryFilter::getInstance($argument);
 
+        if ($argName != $key) {
```
Comment (spawnia) [inline src/Support/Traits/HandlesQueryFilter.php]: How about we always do this and eliminate this check here and the check in QueryFilter.php?

**Comment by spawnia** [comment]
Looks good overall, i think we might improve on simplicity a bit. The test is great, and once the questions are out of the way, i would be happy to see this merged.

**Comment by spawnia** [comment]
LGTM can be merged!

**Comment by chrissm79** [comment]
@hosmelq Thank you for your work on this, good stuff!!

**Pull request merged**